contracts/FeralfileEnglishAuction.sol

```diff
@@ -0,0 +1,334 @@
+        }
+        // emit new bid event
+        emit NewBid(auctionID_, msg.sender, msg.value, false);
+    }
+
+    function placeSignedBid(
+        uint256 auctionID_,
+        address bidder_,
+        uint256 amount_,
+        uint256 timestamp_,
+        bytes32 r_,
+        bytes32 s_,
+        uint8 v_
+    ) external {
+        // get the auction object
+        Auction memory auction_ = auctions[auctionID_];
+        // get the last auction bid object
+        Bid memory highestBid_ = highestBids[auctionID_];
+        // verify the bid is valid
+        _validateBidRequest(
+            auction_,
+            highestBid_,
+            bidder_,
+            amount_,
+            timestamp_
+        );
+
+        // check the signature of (aucId + bidder + amount + timestamp) to make sure the request is from Feral File
+        bytes32 message_ = keccak256(
+            abi.encode(block.chainid, auctionID_, bidder_, amount_, timestamp_)
+        );
+        require(
+            isValidSignature(message_, r_, s_, v_),
+            "FeralfileEnglishAuction: invalid signature"
+        );
+        // transfer last winning bid amount to last bidder if fromFeralFile is false
+        if (!highestBid_.fromFeralFile) {
+            payable(highestBid_.bidder).transfer(highestBid_.amount);
+        }
+        // replace the new bid to bid map
+        highestBids[auctionID_] = Bid({
+            bidder: bidder_,
+            amount: amount_,
+            fromFeralFile: true
+        });
+        // if the gap to end time is lower than threshold, update the auction end time to now() + extend duration
+        if (auction_.endAt - block.timestamp <= auction_.extendThreshold) {
+            auctions[auctionID_].endAt =
+                block.timestamp +
+                auction_.extendDuration;
+        }
+        // emit new bid event
+        emit NewBid(auctionID_, bidder_, amount_, true);
+    }
+
+    function settleAuction(
+        uint256 auctionID_,
+        address tokenAddr_,
+        address vaultAddr_,
+        SaleData memory saleData_,
+        bytes32 r_,
+        bytes32 s_,
+        uint8 v_
+    ) external onlyOwner {
+        // transfer winning bid amount to Feral File Vault contract if winning bid is crypto bid
+        Auction memory auction_ = auctions[auctionID_];
+        require(auction_.id != 0, "FeralfileEnglishAuction: auction not found");
+
+        require(
+            auction_.endAt <= block.timestamp,
+            "FeralfileEnglishAuction: auction not ended"
+        );
+
+        Bid memory highestBid_ = highestBids[auctionID_];
+        require(
+            highestBid_.bidder != address(0),
+            "FeralfileEnglishAuction: no bid"
+        );
+
+        require(
+            !auction_.isSettled,
+            "FeralfileEnglishAuction: auction already settled"
+        );
+
+        auctions[auctionID_].isSettled = true;
+
+        // saleData_.price = highestBid_.amount;
```

Comment:
Why is it commented?

Reply:
accept finalize auction with price is zero